clients/src/main/java/org/apache/kafka/common/protocol/types/FieldDef.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.common.protocol.types;
+
+public class FieldDef {
+    public final String name;
+    public final String docString;
+    public final Type type;
+    public final boolean hasDefaultValue;
+    public final Object defaultValue;
+
+    private FieldDef(String name, String docString, Type type, boolean hasDefaultValue, Object defaultValue) {
+        this.name = name;
+        this.docString = docString;
+        this.type = type;
+        this.hasDefaultValue = hasDefaultValue;
+        this.defaultValue = defaultValue;
+    }
+
+    public static class Int8 extends FieldDef {
```

Comment:
Is there a reason for these to be classes instead of methods? They don't seem to add any behaviour.

Reply:
Making them classes allowed me to overload the `get` and `set` methods of `Struct` with the right type.